instrumentation/opentelemetry-instrumentation-httpx/src/opentelemetry/instrumentation/httpx/__init__.py

```diff
@@ -0,0 +1,325 @@
+    in to the instance.
+    """
+    # pylint:disable=unused-argument
+    def instrumented_sync_init(wrapped, instance, args, kwargs):
+        if context.get_value("suppress_instrumentation"):
+            return wrapped(*args, **kwargs)
+
+        transport = kwargs.get("transport") or httpx.HTTPTransport()
+        telemetry_transport = SyncOpenTelemetryTransport(
+            transport,
+            tracer_provider=tracer_provider,
+            span_callback=span_callback,
+            name_callback=name_callback,
+        )
+
+        kwargs["transport"] = telemetry_transport
+        return wrapped(*args, **kwargs)
+
+    def instrumented_async_init(wrapped, instance, args, kwargs):
+        if context.get_value("suppress_instrumentation"):
+            return wrapped(*args, **kwargs)
+
+        transport = kwargs.get("transport") or httpx.AsyncHTTPTransport()
+        telemetry_transport = AsyncOpenTelemetryTransport(
+            transport,
+            tracer_provider=tracer_provider,
+            span_callback=span_callback,
+            name_callback=name_callback,
+        )
+
+        kwargs["transport"] = telemetry_transport
+        return wrapped(*args, **kwargs)
+
+    wrapt.wrap_function_wrapper(
+        httpx.Client, "__init__", instrumented_sync_init
+    )
+
+    wrapt.wrap_function_wrapper(
+        httpx.AsyncClient, "__init__", instrumented_async_init
+    )
+
+
+def _uninstrument() -> None:
+    """Disables instrumenting for all newly created Client and AsyncClient instances"""
+    unwrap(httpx.Client, "__init__")
+    unwrap(httpx.AsyncClient, "__init__")
+
+
+def _uninstrument_client(
+    client: typing.Union[httpx.Client, httpx.AsyncClient]
+) -> None:
+    """Disables instrumentation for the given Client or AsyncClient"""
+    # pylint: disable=protected-access
+    telemetry_transport: typing.Union[
+        SyncOpenTelemetryTransport, AsyncOpenTelemetryTransport
+    ] = client._transport
+    client._transport = telemetry_transport._transport
+
+
+class HTTPXClientInstrumentor(BaseInstrumentor):
+    """An instrumentor for httpx Client and AsyncClient
+
+    See `BaseInstrumentor`
+    """
+
+    def _instrument(self, **kwargs):
+        """Instruments httpx Client and AsyncClient
+
+        Args:
+            **kwargs: Optional arguments
+                ``tracer_provider``: a TracerProvider, defaults to global
+                ``span_callback``: A callback provided with the response info
+                    to modify the span
+                ``name_callback``: A callback provided with the method and url
+                    to process the span name
+        """
+        _instrument(
+            tracer_provider=kwargs.get("tracer_provider"),
+            span_callback=kwargs.get("span_callback"),
+            name_callback=kwargs.get("name_callback"),
+        )
+
+    def _uninstrument(self, **kwargs):
+        _uninstrument()
+
+    @staticmethod
+    def uninstrument_client(
```

Comment:
we'll also need an `instrument_client` to instrument a specific client. We should change `_instrument` to instrument everything (all instances created before and after the instrument function is called).

Reply:
`instrument_client` is now available and the `_instrument` was updated.